src/wallet/wallet2.cpp

```diff
@@ -4236,6 +4410,329 @@ static uint32_t get_count_above(const std::vector<wallet2::transfer_details> &tr
+  request.view_key = string_tools::pod_to_hex(get_account().get_keys().m_view_secret_key);
+  // Always create account if it doesnt exist.
+  request.create_account = true;
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/login", request, response, m_http_client, std::chrono::seconds(5), "POST");
+  m_daemon_rpc_mutex.unlock();
+  THROW_WALLET_EXCEPTION_IF(!r, error::no_connection_to_daemon, "login");
+
+  new_address = response.new_address;
+  MDEBUG("Status: " << response.status);
+  MDEBUG("Reason: " << response.reason);
+  MDEBUG("New wallet: " << response.new_address);
+  return true;
+}
+
+bool wallet2::light_wallet_import_wallet_request(cryptonote::COMMAND_RPC_IMPORT_WALLET_REQUEST::response &response)
+{
+  MDEBUG("Light wallet import wallet request");
+  cryptonote::COMMAND_RPC_IMPORT_WALLET_REQUEST::request oreq;
+  oreq.address = get_account().get_public_address_str(m_testnet);
+  oreq.view_key = string_tools::pod_to_hex(get_account().get_keys().m_view_secret_key);
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/import_wallet_request", oreq, response, m_http_client, std::chrono::seconds(5), "POST");
+  m_daemon_rpc_mutex.unlock();
+  THROW_WALLET_EXCEPTION_IF(!r, error::no_connection_to_daemon, "import_wallet_request");
+
+
+  return true;
+}
+
+void wallet2::light_wallet_get_unspent_outs()
+{
+  
+  MDEBUG("Getting unspent outs");
+  
+  cryptonote::COMMAND_RPC_GET_UNSPENT_OUTS::request oreq;
+  cryptonote::COMMAND_RPC_GET_UNSPENT_OUTS::response ores;
+  
+  oreq.amount = "0";
+  oreq.address = get_account().get_public_address_str(m_testnet);
+  oreq.view_key = string_tools::pod_to_hex(get_account().get_keys().m_view_secret_key);
+
+  m_daemon_rpc_mutex.lock();
+  bool r = epee::net_utils::invoke_http_json("/get_unspent_outs", oreq, ores, m_http_client, std::chrono::seconds(5), "POST");
+  m_daemon_rpc_mutex.unlock();
+  THROW_WALLET_EXCEPTION_IF(!r, error::no_connection_to_daemon, "get_unspent_outs");
+  
+
+  
+  std::unordered_map<crypto::hash,bool> transfers_txs;
+  for(const auto &t: m_transfers)
+    transfers_txs.emplace(t.m_txid,t.m_spent);
+  
+  MDEBUG("FOUND " << ores.outputs.size() <<" outputs");
+  
+  for (const auto &o: ores.outputs) {
+    bool spent = false;
+    bool add_transfer = true;
+    crypto::key_image unspent_key_image;
+    crypto::public_key tx_public_key = AUTO_VAL_INIT(tx_public_key);
+    string_tools::hex_to_pod(o.tx_pub_key, tx_public_key);
+    
+    for (const std::string &ski: o.spend_key_images) {
+      spent = false;
+
+      // Check if key image is for real
+      crypto::key_image calculated_key_image;
+      cryptonote::keypair in_ephemeral;
+      //TODO: Add validation: key_imageIsNotNull
+      string_tools::hex_to_pod(ski, unspent_key_image);
+
+      cryptonote::generate_key_image_helper(get_account().get_keys(), tx_public_key, o.index, in_ephemeral, calculated_key_image);
```

Comment:
These two functions seem to be called a lot more than necessary. A first pass to cache that would help (though it might not be relevant in practice, I don't know...)